extension/chrome/settings/modules/keyserver.ts

```diff
@@ -113,17 +113,19 @@ Catch.try(async () => {
     }));
 
     $('.action_remove_alias').click(Ui.event.prevent('double', async self => {
-      const { addresses } = await Store.getAcct(acctEmail, ['addresses']);
-      await Store.setAcct(acctEmail, { 'addresses': Value.arr.withoutVal(addresses || [], $(self).attr('email')!) });
+      const { sendAs } = await Store.getAcct(acctEmail, ['sendAs']);
+      if (sendAs) {
+        delete sendAs[$(self).attr('email')!];
+        await Store.setAcct(acctEmail, { sendAs });
+      }
       window.location.reload();
     }));
 
     const contentEl = Xss.sanitizeAppend('#content', '<div class="line"><a href="#" class="action_fetch_aliases">Missing email address? Refresh list</a></div>');
     contentEl.find('.action_fetch_aliases').click(Ui.event.prevent('parallel', async (self, done) => {
       Xss.sanitizeRender(self, Ui.spinner('green'));
       try {
-        const addresses = await Settings.fetchAcctAliasesFromGmail(acctEmail);
-        await Store.setAcct(acctEmail, { addresses: Value.arr.unique(addresses.concat(acctEmail)) });
+        await Settings.refreshAcctAliases(acctEmail);
```

Comment:
Since we're at it, you can remove this button and also the handler. Please use the same logic as in compose - load sendAs asynchronously on every page load, and if it changed, offer to reload the page.

Fewer buttons to confuse users. Thanks!

Reply:
Yes, Will do